### ISSUE TITLE
Some issues with buttons

### ISSUE THREAD

**Issue by virtyaluk** [google/material-design-lite#1864]
Hi, guys!

I love the way you make MDL but, unfortunately, I've found some issues during play with the project. So, I decided to share them with you.

Buttons have some inconsistencies with the official guidance:

1) As the [guidance](https://www.google.com/design/spec/components/buttons.html#buttons-flat-raised-buttons) says, the minimum width of a button should be `88px` but MDL uses the incorrect value of `64px`. The value used by MDL is an appropriate value for *dialog buttons*.
To fix this, open **_variables.scss** file and change incorrect value:
```sass
$button-min-width: 64px !default; // Here should be 88px
````

2) Disabled raised buttons should not have any shadow.

![Alt](https://pp.vk.me/c622418/v622418370/567b5/SyJWFF-2098.jpg)
Left - Google docs, right - MDL.
To fix this, open **_button.scss** file, scroll to "disabled" section and remove following lines:
````sass
@include shadow-2dp(); // This should be removed
````

3) While button (flat or icon) remains in hover state its `background-color` should be `20%` of actual text `color`. This behavior is particularly described in [Icon toggles](https://www.google.com/design/spec/components/buttons.html#buttons-toggle-buttons) section of official guidance but, I suppose, it's a valid behavior for flat button either. But MDL does not implemented it at all while other Material based frameworks do.
![Alt](https://pp.vk.me/c622418/v622418370/567bc/ohtQpYvYkao.jpg)

And, in the end, few little problems I've paid attention to.

1) If you look closer at the button you will see that the text content a little aligned to the top of the button. It's a bit annoying. The reason why buttons behave in this way is because MDL uses `line-height` value of `36px` - the `height` of button itself. So, if you change `line-height`, for example, to `1px` - it fixes the issue:
````sass
line-height: $button-height; // I beg you fix me!
````

2) And finally, icon button's ripple size.

MDL uses very strange values for ripple size within icon button. I've inspected other Material based frameworks and they use all the same values for it. MDL icon button ripples so small so from time to time I've asked myself "What's that?" and "Why it here?".
Though, official guidance has no any word about ripple size, I've created this picture based on "Icon toggles" guidance and the values I got while inspecting MDL and other Material based frameworks.
![Alt](https://pp.vk.me/c622418/v622418370/567c3/IYh-byBEjuc.jpg)
Blue rect - `24px` by `24px`, the canonical size of icon in Material Design.
Red rect - `32px` by `32px`, the size of icon button ripple in MDL.
Green rect - `48px` by `48px`, the real size for ripple in official docs and all other frameworks.

So, here's the question: Why don't you use normal (big, real, whatever) size of ripple? It's looks and feels more natural.

Thanks for your attention and sorry for that wall of text:)

**Comment by virtyaluk** [comment]
Maybe there's a way to fix the bugs (marked by moderator) and PR by myself?

**Comment by Garbee** [comment]
It would be great if individual issues were made for these. Some are no-brainers while others need some discussion since the button spec itself is showing different measurements in different places.

**Comment by virtyaluk** [comment]
@Garbee
I understand that.
Will you review these or do I really need to split these into individual issues?

P.S: I've found some issues with Data tables & selectors, so these new will be added as separate issues:)

**Comment by Garbee** [comment]
I'd much prefer separate issues for everything. That way all discussion is clear and direct. This way we aren't reviewing walls of text with possibly more walls of text creating a cycle which is also confusing.

With separate issues, we can also mark the specific bits that need guidance from the MD team and it is easier for them to look at. Marking this as it is, it is vague as to what they need to focus on and I'd rather ease their efforts trying to improve the specification.

**Comment by virtyaluk** [comment]
@Garbee 
Ok, sorry for the inconvenience.

I'm going to split it a little bit later.

**Comment by Garbee** [comment]
Thank you. We can leave this open until new issues are opened for each individual issue. Sorry for having to insist so much, but it will make things much easier for getting certain spec things discussed.

**Comment by samccone** [comment]
Hey @virtyaluk thanks for the A+ bug report, this level of detail is amazing and makes it really easy for us to investigate and fix the problems. We really appreciate the depth of this report and the effort you put into each explanation here, again much thanks.

I have gone ahead and split up the issue into smaller issues so that we can address each one in parallel. I think we may just want to leave this issue open until, we come to a resolution across the board.


:+1: :clap: Thanks again @virtyaluk

**Comment by sgomes** [comment]
Looks like we have all of the separate issues open up, so closing this one. Thanks. @virtyaluk!
